src/main/java/com/nike/cerberus/cache/DatabaseCache.java

```diff
@@ -0,0 +1,237 @@
+   * @return The number of reads that must be exceeding while counts are being chained before caching of that object is enabled.
+   */
+  protected int getRepeatReadThreshold(Config config, String mapperKey) {
+    int globalRepeatReadThreshold = config.hasPath(GLOBAL_REPEAT_READ_THRESHOLD) ? config.getInt(GLOBAL_REPEAT_READ_THRESHOLD) : DEFAULT_REPEAT_READ_THRESHOLD;
+    String repeatReadThresholdOverridePath = String.format(REPEAT_READ_THRESHOLD_OVERRIDE_PATH_TEMPLATE, mapperKey);
+    return config.hasPath(repeatReadThresholdOverridePath) ? config.getInt(repeatReadThresholdOverridePath) : globalRepeatReadThreshold;
+  }
+
+  @Override
+  public String getId() {
+    return id;
+  }
+
+  @Override
+  public void putObject(Object key, Object value) {
+    if (key == null || value == null) {
+      return;
+    }
+
+    // If the read counter exists and is greater than the threshold then we are receiving
+    // burst repeat reads and we will cache that entry.
+    ofNullable(autoExpiringRepeatReadCounterMap.getIfPresent(key)).ifPresent(counter -> {
+      if (counter.getCount() > repeatReadThreshold) {
+        dataCache.put(key, value);
+      }
+    });
+  }
+
+  @Override
+  public Object getObject(Object key) {
+    Object value = dataCache.getIfPresent(key);
+    if (value == null) {
+      missCounter.inc();
+    } else {
+      hitCounter.inc();
+    }
+
+    // Increment the read counter, which resets after counterExpireTimeInSeconds.
+    Counter counter = autoExpiringRepeatReadCounterMap.getIfPresent(key);
+    if (counter != null) {
+      counter.inc();
+    } else {
+      counter = new Counter();
+      counter.inc();
+      autoExpiringRepeatReadCounterMap.put(key, counter);
+    }
+
+    return value;
+  }
+
+  @Override
+  public Object removeObject(Object key) {
+    Object res = dataCache.getIfPresent(key);
+    dataCache.invalidate(key);
+    return res;
+  }
+
+  @Override
+  public void clear() {
+    // NO-OP, my batis by default clears the entire namespaced cache when a write action occurs,
+    // we do not want that here, we are expiring the cache / making reads eventually consistent.
+    // Since we run Cerberus in a cluster anyways and each instance will have it's own generated cache, a simple small
+    // time window where items purge themselves is adequate.
+  }
+
+  @Override
+  public int getSize() {
+    try {
+      return Math.toIntExact(dataCache.estimatedSize());
+    } catch (ArithmeticException e) {
+      return Integer.MAX_VALUE;
+    }
+  }
+
+  @Override
+  public ReadWriteLock getReadWriteLock() {
+    return null;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(dataCache, getId());
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (!(o instanceof DatabaseCache)) return false;
+    DatabaseCache that = (DatabaseCache) o;
+    return dataCache.equals(that.dataCache) &&
+        getId().equals(that.getId());
```

Comment:
Is getId().equals(that.getId()) necessary?

Reply:
This is generated code.

Reply:
![image](https://user-images.githubusercontent.com/711726/67247777-3d37c100-f417-11e9-9d6f-2be4ac3ae2ac.png)